core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -1742,4 +1753,129 @@ public void testCatalogTokenRefreshDisabledWithCredential() {
             eq(catalogHeaders),
             any());
   }
+
+  @Test
+  public void multipleDiffsAgainstSingleTable() {
+    Namespace namespace = Namespace.of("namespace");
+    TableIdentifier identifier = TableIdentifier.of(namespace, "multipleDiffsAgainstSingleTable");
+
+    Transaction transaction = catalog().buildTable(identifier, SCHEMA).create().newTransaction();
+
+    UpdateSchema updateSchema =
+        transaction.updateSchema().addColumn("new_col", Types.LongType.get());
+    Schema expectedSchema = updateSchema.apply();
+    updateSchema.commit();
+
+    UpdatePartitionSpec updateSpec =
+        transaction.updateSpec().addField("shard", Expressions.bucket("id", 16));
+    PartitionSpec expectedSpec = updateSpec.apply();
+    updateSpec.commit();
+
+    TableCommit tableCommit =
+        ImmutableTableCommit.builder()
+            .identifier(identifier)
+            .base(((BaseTransaction) transaction).startMetadata())
+            .changes(((BaseTransaction) transaction).currentMetadata().changes())
+            .build();
+
+    restCatalog.commitTransaction(ImmutableList.of(tableCommit));
+
+    Table loaded = catalog().loadTable(identifier);
+    assertThat(loaded.schema().asStruct()).isEqualTo(expectedSchema.asStruct());
+    assertThat(loaded.spec().fields()).isEqualTo(expectedSpec.fields());
+  }
+
+  @Test
+  public void multipleDiffsAgainstMultipleTables() {
+    Namespace namespace = Namespace.of("multiDiffNamespace");
+    TableIdentifier identifier1 = TableIdentifier.of(namespace, "multiDiffTable1");
+    TableIdentifier identifier2 = TableIdentifier.of(namespace, "multiDiffTable2");
+
+    Transaction transaction1 = catalog().buildTable(identifier1, SCHEMA).create().newTransaction();
+    Transaction transaction2 = catalog().buildTable(identifier2, SCHEMA).create().newTransaction();
+
+    UpdateSchema updateSchema =
+        transaction1.updateSchema().addColumn("new_col", Types.LongType.get());
+    Schema expectedSchema = updateSchema.apply();
+    updateSchema.commit();
+
+    UpdateSchema updateSchema2 =
+        transaction2.updateSchema().addColumn("new_col2", Types.LongType.get());
+    Schema expectedSchema2 = updateSchema2.apply();
+    updateSchema2.commit();
+
+    TableCommit tableCommit1 =
+        ImmutableTableCommit.builder()
+            .identifier(identifier1)
+            .base(((BaseTransaction) transaction1).startMetadata())
+            .changes(((BaseTransaction) transaction1).currentMetadata().changes())
+            .build();
+
+    TableCommit tableCommit2 =
+        ImmutableTableCommit.builder()
+            .identifier(identifier2)
+            .base(((BaseTransaction) transaction2).startMetadata())
+            .changes(((BaseTransaction) transaction2).currentMetadata().changes())
+            .build();
+
+    restCatalog.commitTransaction(ImmutableList.of(tableCommit1, tableCommit2));
```

Comment:
To make things like this easier, we tend to add convenience methods, like `commitTransaction(TableCommit... commits)` with a default that creates the `ImmutableList`. That makes tests a bit easier to read and write.